common/src/main/java/com/datastax/oss/common/sink/config/AsciiDocGenerator.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.common.sink.config;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Path;
+import java.util.*;
+import org.apache.kafka.common.config.ConfigDef;
+
+/** Generates Asiidoc documentation for a ConfigDef */
+public class AsciiDocGenerator {
+
+  public static String toAscidoc(ConfigDef configDef) {
+    StringBuilder b = new StringBuilder();
+    for (ConfigDef.ConfigKey key : sortedConfigs(configDef)) {
+      if (key.internalConfig) {
+        continue;
+      }
+      getConfigKeyAsciiDoc(configDef, key, b);
+      b.append("\n");
+    }
+    return b.toString();
+  }
+
+  protected static void getConfigKeyAsciiDoc(
+      ConfigDef configDef, ConfigDef.ConfigKey key, StringBuilder b) {
+    b.append("[#").append(key.name).append("]").append("\n");
+    for (String docLine : key.documentation.split("\n")) {
+      if (docLine.length() == 0) {
+        continue;
+      }
+      b.append(docLine).append("\n+\n");
+    }
+    b.append("Type: ").append(getConfigValue(key, "Type")).append("\n");
+    if (key.hasDefault()) {
+      b.append("Default: ").append(getConfigValue(key, "Default")).append("\n");
+    }
+    if (key.validator != null) {
+      b.append("Valid Values: ").append(getConfigValue(key, "Valid Values")).append("\n");
+    }
+    b.append("Importance: ").append(getConfigValue(key, "Importance")).append("\n");
+  }
+
+  protected static String getConfigValue(ConfigDef.ConfigKey key, String headerName) {
+    switch (headerName) {
+      case "Name":
```

Comment:
Couldn't you use an enum here?

Reply:
Maybe, but this code come from the RST doc generator from Kafka, already well tested. We just need to duplicate such function because there were private or protected in the Kafka package.